lib/spack/spack/binary_distribution.py

```diff
@@ -1139,6 +1139,66 @@ def generate_key_index(key_prefix, tmpdir=None):
                 shutil.rmtree(tmpdir)
 
 
+@contextmanager
+def gzip_compressed_tarfile(path):
+    """Create a reproducible, compressed tarfile"""
+    # Create gzip compressed tarball of the install prefix
+    # 1) Use explicit empty filename and mtime 0 for gzip header reproducibility
+    # 2) On AMD Ryzen 3700X and an SSD disk, we have the following on compression speed:
+    # compresslevel=6 gzip default: llvm takes 4mins, roughly 2.1GB
+    # compresslevel=9 python default: llvm takes 12mins, roughly 2.1GB
+    # So we follow gzip.
+    with open(path, "wb") as fileobj, closing(
+        GzipFile(filename="", mode="wb", compresslevel=6, mtime=0, fileobj=fileobj)
```

Comment:
Why do we set `filename` here? Wouldn't it be sufficient to:
```suggestion
        GzipFile(fileobj=fileobj, mode="wb", compresslevel=6, mtime=0)
```
?

Reply:
Update: discussed in DMs. This is needed to avoid using `fileobj.name`

https://github.com/python/cpython/commit/68de37973f9b2bd01ba91b10e01fb207396b145f

Reply:
@haampie This LGTM. I'll merge as soon as we add a comment on this nitpick, to avoid forgetting why it was necessary in the future.